scipy/signal/lti_conversion.py

```diff
@@ -384,6 +384,36 @@ def cont2discrete(system, dt, method="zoh", alpha=None):
     approximation is based on [2]_ and [3]_, the First-Order Hold (foh) method
     is based on [4]_.
 
+    Examples
+    --------
+    We can transform a continuous state-space system to a discrete one:
+
+    >>> import matplotlib.pyplot as plt
+    >>> from scipy.signal import cont2discrete, lti, dlti, dstep
+
+    Define a continious state-space system.
+
+    >>> A = np.array([[0, 1],[-10, -3]])
+    >>> B = np.array([[0],[10]])
+    >>> C = np.array([[1, 0]])
+    >>> D = np.array([[0]])
+    >>> l_system = lti(A, B, C, D)
+    >>> t, x = l_system.step(T=np.linspace(0,5,100))
+    >>> plt.plot(t, x, label='Continuous')
+    >>> plt.hlines(1, t[0], t[-1], linestyles='dotted')
+
+    Transform it to a discrete state-space system using several methods.
+
+    >>> dt = 0.1
```

Comment:
They didn't push too hard but Matplotlib folks promote what they call the OO interface as such also I've added the linewidth to the continuous case for emphasis.

```python
fig, ax = plt.subplots()
ax.plot(t, x, label='Continuous', linewidth=3)
ax.hlines(1, t[0], t[-1], linestyles='dotted')
dt = 0.1
for method in ['zoh', 'bilinear', 'euler', 'backward_diff', 'foh', 'impulse']:
    d_system = cont2discrete((A, B, C, D), dt, method=method)
    s, x_d = dstep(d_system)
    ax.step(s, np.squeeze(x_d), label=method, where='post')
ax.axis([t[0], t[-1], x[0], 1.4])
ax.legend(loc='best')
fig.tight_layout()
plt.show()
```

Reply:
Good point, we should keep this in mind for some large refactoring and other PRs like that. Maybe it would be worth at least commenting on the big issue in case people are not yet aware.

Reply:
OK. I used OO interfaces.